### ISSUE TITLE
not working on android 8.1

### ISSUE THREAD

**Issue by duub** [tobykurien/BatteryFu#44]
has anyone tested on android 8.1? I'm using lineageOS 15.1 (unnoficial) with aamsung galaxy s6 and is not disabling the data connection

**Comment by tobykurien** [comment]
Yup, as far as I know, mobile data toggling is no longer supported by the OS. Only WiFi toggle will work.

**Comment by duub** [comment]
I'm not an Android developer... I only know the programming basics...  to put in context... I've been reading your code and I can see in MobileDataSwitcher that there are different classes for some of android releases... Maybe there is a way to fix it creating and OreoSwitch with the function calls comented here:
https://stackoverflow.com/questions/31120082/latest-update-on-enabling-and-disabling-mobile-data-programmatically

Here there's the reference to the library referenced in the second answer but updated to android 8.1: http://androidxref.com/8.1.0_r33/xref/frameworks/opt/telephony/src/java/com/android/internal/telephony/PhoneInternalInterface.java

thanks for readind this, I find your app very usefull!

**Comment by ioctl-user** [comment]
> Yup, as far as I know, mobile data toggling is no longer supported by the OS. Only WiFi toggle will work.

It's wrong. Mobile data toggling work in LineageOS 17.1 (Android 10).

**Comment by Korb** [comment]
Currently when viewing [an BatteryFu screen in F-Droid](https://f-droid.org/packages/com.tobykurien.batteryfu/) I see the following notification:

> This app was built for an older version of Android and cannot be updated automatically.

OS version: Funtouch OS 14
Model: Vivo T1 (4G, V2153)
Hardware version: MP_0.1
Build number: PD22010F_EX_A_14.2.9.0.W20
Baseband version: .HA.1.1.c2-00054-DIVAR_GENSP_PACK-1.29062.107
Kernel version: 4.19.157-perf+, 1 Sat Oct 12 11:14:50 CST 2024
Compile time: 12 Oct 2024 06:06:30
Android security update: 1 October 2024
Google Play system update: 1 October 2024
Software parameter update: 29 October 2024

I suggest you read the article "[Meet Google Play's target API level requirement](https://developer.android.com/google/play/requirements/target-sdk)", which is primarily intended for Google Play users, but F-Droid users will also have difficulty finding, installing, and updating the app in situations where OS security requirements conflict with the user app.

Also, I recommend keeping in mind [the changes coming in 2025](https://android-developers.googleblog.com/2024/10/android-sdk-release-update.html).
